### PR TITLE
TxDownloadManager followups

### PR DESCRIPTION
**Pull request by glozow**
Addresses some remaining followups from #30110:
- https://github.com/bitcoin/bitcoin/pull/30110#discussion_r1818893833
- https://github.com/bitcoin/bitcoin/pull/30110#discussion_r1819638959
- https://github.com/bitcoin/bitcoin/pull/30110#discussion_r1819634235

### CODE REVIEW

**Comment by instagibbs** [comment]
is this ready for undraft?

**Review by naumenkogs — COMMENTED** [review]
ACK 8351562bec6081eda2a600bfe4edeb264a9dee0b

I haven't verified what happens if `time` goes below 0

**Comment by dergoegge** [comment]
oss-fuzz found a crash in txdownloadman_impl, perhaps worth including the fix here or in another followup:

```
$ FUZZ=txdownloadman_impl out/libfuzzer/fuzz clusterfuzz-testcase-minimized-txdownloadman_impl-5180010463297536.txt
src/test/fuzz/txdownloadman.cpp:290 CheckInvariants: Assertion `txdownload_impl.m_txrequest.CountInFlight(peer) <= node::MAX_PEER_TX_REQUEST_IN_FLIGHT' failed.
```

[clusterfuzz-testcase-minimized-txdownloadman_impl-5180010463297536.txt](https://github.com/user-attachments/files/17645609/clusterfuzz-testcase-minimized-txdownloadman_impl-5180010463297536.txt)

**Comment by dergoegge** [comment]
> is that on master

Sorry I should have clarified, it is on master.

**Comment by glozow** [comment]
Thanks @dergoegge, I probably also could have inferred from it being oss-fuzz.

Even though I wrote it, the assertion looks wrong 😅 overloaded inflight is handled with just a delay

https://github.com/bitcoin/bitcoin/blob/80cb630bd945e7f1777c7bab041c411b6916e651/src/node/txdownloadman_impl.cpp#L199

I think what I was trying to do was ensure `MAX_PEER_TX_ANNOUNCEMENTS` isn't busted. Adding a fix here.

**Review by instagibbs — APPROVED** [review]
ACK 5dc94d13d419b8d5e543cb50edeb872335c090e7

File: src/test/fuzz/txdownloadman.cpp [hunk 1]
```diff
@@ -287,7 +287,7 @@ static void CheckInvariants(const node::TxDownloadManagerImpl& txdownload_impl,
     // We should never have more than the maximum in-flight requests out for a peer.
     for (NodeId peer = 0; peer < NUM_PEERS; ++peer) {
         if (!HasRelayPermissions(peer)) {
-            Assert(txdownload_impl.m_txrequest.CountInFlight(peer) <= node::MAX_PEER_TX_REQUEST_IN_FLIGHT);
+            Assert(txdownload_impl.m_txrequest.Count(peer) <= node::MAX_PEER_TX_ANNOUNCEMENTS);
```
Comment (instagibbs) [inline src/test/fuzz/txdownloadman.cpp:290]: any value in 
```suggestion
            Assert(txdownload_impl.m_txrequest.CountInFlight(peer) <= node::MAX_PEER_TX_ANNOUNCEMENTS);
            Assert(txdownload_impl.m_txrequest.Count(peer) <= node::MAX_PEER_TX_ANNOUNCEMENTS);
```
?

Comment (glozow) [reply src/test/fuzz/txdownloadman.cpp:290]: What do you mean any value in?

Comment (instagibbs) [reply src/test/fuzz/txdownloadman.cpp:290]: would the suggested change potentially give coverage

**Review by theStack — APPROVED** [review]
ACK 5dc94d13d419b8d5e543cb50edeb872335c090e7

**Review by Joj501 — APPROVED** [review]
Connect with my wallet address

File: src/test/fuzz/txdownloadman.cpp (hunk 1 above)
Comment (naumenkogs) [reply src/test/fuzz/txdownloadman.cpp:290]: Perhaps `CountInFlight` should check against `MAX_PEER_TX_REQUEST_IN_FLIGHT` and `Count` against `MAX_PEER_TX_ANNOUNCEMENTS`, separately?
I think that's what Greg wanted to suggest.

**Comment by naumenkogs** [comment]
ACK https://github.com/bitcoin/bitcoin/commit/5dc94d13d419b8d5e543cb50edeb872335c090e7

File: src/test/fuzz/txdownloadman.cpp (hunk 1 above)
Comment (marcofleon) [reply src/test/fuzz/txdownloadman.cpp:290]: Could be misunderstanding here, but wouldn't `m_txrequest.Count(peer)` always be greater than (or equal?) to `m_txrequest.CountInFlight(peer)`, so the additional assert would be redundant?

File: src/test/fuzz/txdownloadman.cpp [hunk 2]
```diff
@@ -287,7 +287,7 @@ static void CheckInvariants(const node::TxDownloadManagerImpl& txdownload_impl,
     // We should never have more than the maximum in-flight requests out for a peer.
```
Comment (marcofleon) [inline src/test/fuzz/txdownloadman.cpp:287]: Update comment for the new assert?

**Comment by instagibbs** [comment]
Assuming we implemented everything correctly yes!

**Comment by marcofleon** [comment]
> Assuming we implemented everything correctly yes!

hmm... you make a good point

**Pull request merged**